### ISSUE TITLE
Finish New Autocomplete Input

### ISSUE THREAD

**Issue by GalGend** [meirim-org/meirim#443]
-DUPLICATE- will be done by @harshitsan in a parallel task

Finish the integration of the new Plan search bar component from this PR: 
https://github.com/meirim-org/meirim/pull/429

Remaining tasks:

- [ ] Make component sticky.
- [ ]  Close the loop on filter by status (align with the api IN_PROGRESS/ APPROVED status).
- [ ] Use the new component (instead of the current one) in Plans.jsx:

Blocked by #468

**Comment by GalGend** [comment]
closing due to design change
